python/lsst/cp/pipe/ptc.py

```diff
@@ -354,6 +371,26 @@ def runDataRef(self, dataRefList):
             id2 = exp2.getInfo().getVisitInfo().getExposureId()
             expIds.append((id1, id2))
         self.log.info(f"Measuring PTC using {expIds} exposures for detector {detector.getId()}")
+
+        # get photodiode data early so that logic can be put in to only use the
+        # data if all files are found, as partial corrections are not possible
+        # or at least require significant logic to deal with
+        if self.config.doPhotodiode:
+            for (expId1, expId2) in expIds:
+                charges = [-1, -1]  # necessary to have a not-found value to keep lists in step
+                for i, expId in enumerate([expId1, expId2]):
+                    # XXX you really need to remove this bit !!!!!!!!!!!!!!!!!!
+                    dataRef.dataId['expId'] = expId//1000
+                    if self.config.photodiodeDataPath:
+                        photodiodeData = getBOTphotodiodeData(dataRef, self.config.photodiodeDataPath)
+                    else:
+                        photodiodeData = getBOTphotodiodeData(dataRef)
+                    if photodiodeData:  # default path stored in function def to keep task clean
+                        charges[i] = photodiodeData.getCharge()
```

Comment:
Can this warn via the log if `photodiodeData is None`?  It might be helpful to find out early if it'll appear in the output dataset.